### PR TITLE
Add Scala implementation

### PR DESCRIPTION
**Pull request by seblm**
Thank you for this useful kata. With this pull request, I introduce a proposition of Scala implementation. There is also a GitHub Action that ensures that formatting, compiling and testing are always ok.

### CODE REVIEW

File: scala/src/test/scala/characters/ExampleCharacterTestCase.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    // This is the 'act' step
+    val charactersList = finder.findFamilyByLastName(Some("Wheeler"))
+
+    // this is example code showing kinds of assertion you could do on a List of Characters
+    assert(charactersList != null)
+    assertEquals(charactersList.length, 3)
+    assert(charactersList.contains(new Character("Nancy", Some("Wheeler"))))
+    assertEquals(
+      charactersList.sortBy(_.firstName),
```
Comment (emilybache) [inline scala/src/test/scala/characters/ExampleCharacterTestCase.scala]: Is there a test framework you could use that would include an assertion "contains elements" without enforcing a particular order?

**Comment by emilybache** [comment]
Thankyou this looks really good. The desire for a "collection contains these elements" assertion is a nice-to-have - is it possible do you think?

File: scala/src/test/scala/characters/ExampleCharacterTestCase.scala (hunk 1 above)
Comment (seblm) [reply scala/src/test/scala/characters/ExampleCharacterTestCase.scala]: I am not familiar with [munit test framework](https://scalameta.org/munit) introduced by scafolded project `sbt new scala/scala3.g8`. I will change to more conventional [ScalaTest](https://www.scalatest.org) which has a more meaningful assertion library.

**Comment by seblm** [comment]
@emilybache please have a look at 14892f9 that uses ScalaTest and more expressive matchers

**Comment by emilybache** [comment]
Thankyou that looks good!

**Pull request merged**